cobigen-templates/templates-devon4j/src/main/templates/sql_trigger/functions.ftl

```diff
@@ -0,0 +1,74 @@
+  <#elseif field.type?contains("Character") || field.type=="char">
+     <#assign type = "CHAR">
+  <#elseif field.type?contains("Byte") || field.type=='byte'>
+     <#assign type = "TINYINT">                   
+  <#elseif field.type?contains("Boolean") || field.type=="boolean">
+     <#assign type = "BOOLEAN">
+  <#elseif field.type?contains("Instant") || field.type?contains("Timestamp")>
+     <#assign type = "TIMESTAMP">
+  <#elseif field.type?contains("Date") || field.type?contains("Calender")>
+     <#assign type = "DATE">
+  <#elseif field.type?contains("Time")>
+     <#assign type = 'TIME'>  
+  <#elseif field.type?contains("UUID")>
+     <#assign type = "BINARY">  
+  <#elseif field.type?contains("Blob")>
+     <#assign type = "BLOB">        
+  <#elseif field.annotations.javax_validation_constraints_Size?? && field.annotations.javax_validation_constraints_Size.max??>
+     <#assign type = "VARCHAR(" + field.annotations.javax_validation_constraints_Size.max +")">
+  <#else>
+     <#assign type = "VARCHAR(255)">
+  </#if> 
+  <#if (field.annotations.javax_persistence_Column??
+               && field.annotations.javax_persistence_Column.nullable?? 
+               && field.annotations.javax_persistence_Column.nullable=="false") || field.annotations.javax_validation_constraints_NotNull?? >
+        <#assign type = type + " NOT NULL">   
+  </#if>
+</#macro>
+
+<#macro get_mapping_type input_type>
+  <#if input_type?contains("Integer") || input_type=="int" || input_type?contains("Year") || input_type?contains("Month") || JavaUtil.isEnum(input_type)>
+    <#assign type = "INTEGER">
+  <#elseif input_type?contains("Long") || input_type=="long" || input_type?contains("Object")>
+     <#assign type = "BIGINT">
+  <#elseif input_type?contains("Short") || input_type=="short">
+     <#assign type = "SMALLINT">
+  <#elseif input_type?contains("Float") || input_type=="float">
+     <#assign type = "FLOAT">             
+  <#elseif input_type?contains("Double") || input_type=="double">
+     <#assign type = "DOUBLE">
+  <#elseif input_type?contains("BigDecimal") || input_type?contains("BigInteger")>
+     <#assign type = "NUMERIC">
+  <#elseif input_type?contains("Character") || input_type=="char">
+     <#assign type = "CHAR">
+  <#elseif input_type?contains("Byte") || input_type=='byte'>
+     <#assign type = "TINYINT">                   
+  <#elseif input_type?contains("Boolean") || input_type=="boolean">
+     <#assign type = "BOOLEAN">
+  <#elseif input_type?contains("Instant") || input_type?contains("Timestamp")>
+     <#assign type = "TIMESTAMP">
+  <#elseif input_type?contains("Date") || input_type?contains("Calender")>
+     <#assign type = "DATE">
+  <#elseif input_type?contains("Time")>
+     <#assign type = 'TIME'>  
+  <#elseif input_type?contains("UUID")>
+     <#assign type = "BINARY">  
+  <#elseif input_type?contains("Blob")>
+     <#assign type = "BLOB">        
+  <#else>
+     <#assign type = "VARCHAR(255)">
+  </#if> 
+</#macro>
```

Comment:
here I would think macro is a little problematic.
In FreeMarker, you have macros and functions.
Macros are program snippets which will be copied to the place and executed there. No isolation of variables etc. such that it might be hard to reuse macros in any place as variables have to have the same name etc.
So what you want to do here is I think better call a freemarker function which returns to you the string of the type, which you can then assign in your template to a local variable.